pkg/v1/remote/descriptor.go

```diff
@@ -479,3 +492,20 @@ func (f *fetcher) blobExists(h v1.Hash) (bool, error) {
 
 	return resp.StatusCode == http.StatusOK, nil
 }
+
+// If filter applied, filter out by artifactType.
+// See https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers
+func applyFilterToReferrersResponse(filter map[string]string, im *v1.IndexManifest) {
+	if filter == nil {
+		return
+	}
+	if v, ok := filter["artifactType"]; ok {
+		tmp := []v1.Descriptor{}
+		for _, desc := range im.Manifests {
+			if desc.ArtifactType == v {
+				tmp = append(tmp, desc)
+			}
+		}
+		im.Manifests = tmp
```

Comment:
This could also just return `tmp` and line 270 and 290 could be `return applyFilter(...), nil`